libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1087,10 +1091,21 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 	{
 		// No initial value is only permitted for single variables with specified type.
 		if (_statement.declarations().size() != 1 || !_statement.declarations().front())
-			m_errorReporter.fatalTypeError(_statement.location(), "Assignment necessary for type detection.");
+		{
+			if (!boost::algorithm::all_of_equal(_statement.declarations(), nullptr))
+				// Error already reported in syntax checker, but we need to bail out here.
+				m_errorReporter.syntaxError(_statement.location(), "Use of the \"var\" keyword is disallowed.");
+			return true;
+		}
+
 		VariableDeclaration const& varDecl = *_statement.declarations().front();
 		if (!varDecl.annotation().type)
-			m_errorReporter.fatalTypeError(_statement.location(), "Assignment necessary for type detection.");
+		{
+			m_errorReporter.syntaxError(_statement.location(), "Use of the \"var\" keyword is disallowed");
+			// there is nothing else to check, so we can safely early return at this point.
```

Comment:
Again, you should return false here.

Reply:
And actually I think it should be a fatal error - we will be in trouble if the variable is referenced from somewhere.